crates/ast/src/lexer.rs

```diff
@@ -85,10 +85,12 @@ fn lexer<'src>(
         .to_slice()
         .map(Token::Dec);
 
-    let string = just("\"")
-        .ignore_then(any().and_is(just("\"").not()).repeated().to_slice())
-        .then_ignore(just("\""))
-        .map(Token::String);
+    let string = none_of("\\\"")
+        .or(just('\\').ignore_then(just('"')))
```

Comment:
Now that I think about we kind of need to have full escaping behavior e.g. `\\` -> `\`, `\n` -> `<newline_char>`, etc.